contracts/src/v0.8/automation/dev/test/AutomationRegistry2_3.t.sol

```diff
@@ -711,13 +796,108 @@ contract NOPsSettlement is SetUp {
+    vm.startPrank(UPKEEP_ADMIN);
+    usdToken.approve(address(registry), 1e20);
+    registry.addFunds(id, 1e20);
+
+    // manually create a transmit so TRANSMITTERS earn some rewards
+    _transmit(id, registry);
+
+    // TRANSMITTERS have positive balance now
+    // configure the registry to use NEW_TRANSMITTERS
+    _configureWithNewTransmitters(registry, registrar);
+
+    _transmit(id, registry);
+
+    // verify all transmitters have positive balances
+    address[] memory expectedPayees = new address[](6);
+    uint256[] memory expectedPayments = new uint256[](6);
+    for (uint256 i = 0; i < NEW_TRANSMITTERS.length; i++) {
+      (bool active, uint8 index, uint96 balance, uint96 lastCollected, address payee) = registry.getTransmitterInfo(
+        NEW_TRANSMITTERS[i]
+      );
+      assertTrue(active);
+      assertEq(i, index);
+      assertTrue(lastCollected > 0);
+      expectedPayments[i] = balance;
+      expectedPayees[i] = payee;
+    }
+    for (uint256 i = 2; i < TRANSMITTERS.length; i++) {
+      (bool active, uint8 index, uint96 balance, uint96 lastCollected, address payee) = registry.getTransmitterInfo(
+        TRANSMITTERS[i]
+      );
+      assertFalse(active);
+      assertEq(i, index);
+      assertTrue(balance > 0);
+      assertTrue(lastCollected > 0);
+      expectedPayments[2 + i] = balance;
+      expectedPayees[2 + i] = payee;
+    }
+
+    // verify offchain settlement will emit NOPs' balances
+    vm.startPrank(FINANCE_ADMIN);
+
+    // simply expectEmit won't work here because s_deactivatedTransmitters is an enumerable set so the order of these
+    // deactivated transmitters is not guaranteed. To handle this, we record logs and decode data field manually.
+    vm.recordLogs();
+    registry.settleNOPsOffchain();
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+
+    assertEq(entries.length, 1);
+    Vm.Log memory l = entries[0];
+    assertEq(l.topics[0], keccak256("NOPsSettledOffchain(address[],uint256[])"));
+    (address[] memory actualPayees, uint256[] memory actualPayments) = abi.decode(l.data, (address[], uint256[]));
+    assertEq(actualPayees.length, 6);
+    assertEq(actualPayments.length, 6);
+
+    // first 4 payees and payments are for NEW_TRANSMITTERS and they are ordered.
+    for (uint256 i = 0; i < NEW_TRANSMITTERS.length; i++) {
+      assertEq(actualPayees[i], expectedPayees[i]);
+      assertEq(actualPayments[i], expectedPayments[i]);
+    }
+
+    // the last 2 payees and payments for TRANSMITTERS[2] and TRANSMITTERS[3] and they are not ordered
+    assertTrue(
+      (actualPayments[5] == expectedPayments[5] &&
+        actualPayees[5] == expectedPayees[5] &&
+        actualPayments[4] == expectedPayments[4] &&
+        actualPayees[4] == expectedPayees[4]) ||
+        (actualPayments[5] == expectedPayments[4] &&
+          actualPayees[5] == expectedPayees[4] &&
+          actualPayments[4] == expectedPayments[5] &&
+          actualPayees[4] == expectedPayees[5])
+    );
```

Comment:
😆 I think we could find a simpler way to do this... but I also think we've got more important stuff so I'm gonna let it slide! lol